doc/man3/OSSL_PARAM_construct_from_text.pod

```diff
@@ -0,0 +1,167 @@
+=pod
+
+=head1 NAME
+
+OSSL_PARAM_construct_from_text, OSSL_PARAM_allocate_from_text
+- OSSL_PARAM construction utilities
+
+=head1 SYNOPSIS
+
+ #include <openssl/params.h>
+
+ int OSSL_PARAM_construct_from_text(OSSL_PARAM *to,
+                                    const OSSL_PARAM *paramdefs,
+                                    const char *key, const char *value,
+                                    size_t value_n,
+                                    void *buf, size_t *buf_n)
+ int OSSL_PARAM_allocate_from_text(OSSL_PARAM *to,
+                                   const OSSL_PARAM *paramdefs,
+                                   const char *key, const char *value,
+                                   size_t value_n);
+
+=head1 DESCRIPTION
+
+With OpenSSL before version 3.0, parameters were passed down to or
+retrieved from algorithm implementations via control functions.
+Some of these control functions existed in variants that took string
+parameters, for example L<EVP_PKEY_CTX_ctrl_str(3)>.
+
+OpenSSL 3.0 introduces a new mechanism to do the same thing with an
+array of parameters that contain name, value, value type and value
+size (see L<OSSL_PARAM(3)> for more information).
+
+OSSL_PARAM_construct_from_text() takes a control I<key>, I<value> and
+value size I<value_n>, and given a parameter descriptor array
+I<paramdefs>, it converts the value to something suitable for
+L<OSSL_PARAM(3)> and stores that in the buffer I<buf>, and modifies
+the parameter I<to> to match.
+I<buf_n>, if not NULL, will be assigned the number of bytes used in
+I<buf>.
+If I<buf> is NULL, only I<buf_n> will be modified, everything else is
+left untouched, allowing a caller to find out how large the buffer
+should be.
+I<buf> needs to be safely aligned for all kinds of uses, including as
```

Comment:
This sentence seems wrong, but I can't think of an alternative.  It's the _all kinds of uses_ that's upsetting.

Reply:
Why is it upsetting?

Reply:
It just reads badly (to me).  I think it's the _all_.  An octet string param doesn't need to be aligned but this is implying that the buffer would need to be correctly aligned for a _double_ for example.

What about: `I<buf> needs to be correctly aligned for the type of the OSSL_PARAM I<key>`?

Reply:
Okie, that works